portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -39,4 +39,32 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
-}
+
+@Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, "text-input", "");
+    // messagesList.add(text);
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("content", text);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
```

Comment:
All the string constants here (eg. `text-input`, `Comment`, `content`, `/index.html`) should be made class level `private static final String` variables because they have broader implications in the overall app and are/will-be used in other places. That way you can also reuse those same variables when you update the `get` method to read from the datastore.

Reply:
that makes sense - i didn't think of it like that but just changed. thank you!